naviguide.cc

```diff
@@ -67,36 +67,52 @@ typedef struct {
 /* Global variables */
 
 static gbfile* file_in, *file_out;
-static uint16_t nof_wp;
+static uint16_t nof_wp = 0;
 static route_head* rte_head;
 static ng_file_header_t ng_file_header;
 static ng_wp_no_comment_t WPNC;
 static ng_next_wp_t ng_next_wp;
 static char strComment[101];
 
-/* Process options */
-/* wp - process only waypoints */
-/* rte - process as route */
-/* wprte - Process waypoints and route */
-static char* process = NULL;
-static char* reorder = NULL;
-static int process_rte = 1;
-static int reorder_wp = 0;
+/*
+    File reading options
+    wp - read file as waypoints
+    rte - read file as route
+    trk - read file as track
 
-static char temp_short_name[5];
+ */
+static char* read_as = NULL;
+gpsdata_type read_type = unknown_gpsdata;
 
+/*
+    File writing options:
+    First non-empty data type, as a string with the letters:
```

Comment:
Unfortunately, our existing art on some formats are confusing.

It's strongly encouraged that we write any type of data we have and not really use the -w/-t/-r flags to control the writing. Can both exist at the same time in this format? If not, I'd much rather we use the doing_rte/doing_trk style code instead of adding suboptions.